chore(docs): update box readme, remove duplicated features, added box install to the docs

signorecello opened the pull request:
### TL;DR (AI generated)

Simplified the Aztec app creation process by removing sandbox-related functionality and updating documentation.

### What changed?

- Removed sandbox installation, update, and run steps from the `aztec-app` command.
- Updated README files and documentation to reflect the removal of sandbox-related features.
- Changed the command from `npx create-aztec-app` to `npx aztec-app` throughout the project.
- Removed sandbox-related scripts and dependencies.
- Updated the onCreateCommand script in the devcontainer configuration.

### How to test?

1. Run `npx aztec-app` and verify that it only prompts for project options without mentioning sandbox installation.
2. Check that the README files in various boxes (react, vanilla, contract-only) no longer mention sandbox installation or management.
3. Verify that the devcontainer script uses the new `npx aztec-app` command.

### Why make this change?

This change simplifies the Aztec app creation process by focusing solely on project setup and removing the complexity of sandbox management. It allows users to handle sandbox installation and management separately, providing a more streamlined experience for creating Aztec projects.

signorecello committed:
chore(docs): update box readme, remove duplicated features, added box install to the docs

signorecello commented:
* **#8254** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/8254?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @signorecello and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

catmcgee commented:
should this work? getting `zlib: unexpected end of file` on all options

```
○ → npx aztec-app
? Please choose your type of project: Boilerplate project with frontend
? Please choose your Aztec boilerplate: React App
? Your box name: my-aztec-box
✖ Cloning the box code...
zlib: unexpected end of file
```

signorecello commented:
> should this work? getting `zlib: unexpected end of file` on all options
> 
> ```
> ○ → npx aztec-app
> ? Please choose your type of project: Boilerplate project with frontend
> ? Please choose your Aztec boilerplate: React App
> ? Your box name: my-aztec-box
> ✖ Cloning the box code...
> zlib: unexpected end of file
> ```
> 
> 

Sorry I should've put the repro steps in the description:

- Check out the branch
- Head to `boxes`
- Run `npx .` and the rest of the command

In any case `aztec-app` shouldn't break like it did for you :D So I'll investigate

signorecello committed:
wip (tests)

signorecello committed:
chore(boxes): add test for boxes

critesjosh commented:
What is the --github_token option for?

critesjosh reviewed (changes requested):
When I install the contract example, I still get a `package.json` and a `tsconfig.json` that are irrelevant for a simple contract example. Can you update so that unnecessary files are not installed?

signorecello commented:
> What is the --github_token option for?

It's optional, you can pass it to avoid rate limiting (i.e. for testing)

signorecello commented:
> When testing the React box (UPDATE: got the same error with vanilla UI), I got an error:
> 
> ![image](https://github.com/user-attachments/assets/bfdc944f-fc85-4dbd-8392-bac95b7d5373)
> 
> 
> `yarn add assert` and adding the fallback
> ```js
>       assert: require.resolve('assert/'),
> ```
> to webpack.config.js [here](https://github.com/AztecProtocol/aztec-packages/blob/1afc533eed0238fa7fef9c284079bb955ba4a004/boxes/boxes/react/webpack.config.js#L39) resolved it. Try it out on your side and see if it fixes it

Need to check why wasn't this caught in the CI testing...

signorecello committed:
bumping playwright so it works with ubuntu 22.04, adding the missing polyfill, and removing unnecessary files from contract box

signorecello committed:
committing lockfile

signorecello committed:
bumping playwright

signorecello committed:
yarn

signorecello committed:
Merge branch 'master' into zpedro/boxes_readme

critesjosh reviewed (changes requested):
I am still getting a `package.json` and `.yarnrc.yml` file in project when running `npx init`. Is this intentional? 

![image](https://github.com/user-attachments/assets/f7e08bff-e1fb-4c03-82f9-8b1e15da6220)

Also I am still seeing the following error. I am on your branch, running `npx . new` and moving into the created directory. Is there something else that needs updating?

![image](https://github.com/user-attachments/assets/cf9b7f50-c3b7-462e-a64a-406377806638)

signorecello commented:
> I am still getting a `package.json` and `.yarnrc.yml` file in project when running `npx init`. Is this intentional?

I'm an idiot and removed those files from `contract_only` boxes instead of the `init`... ** breathes deeply **

> Also I am still seeing the following error. I am on your branch, running npx . new and moving into the created directory. Is there something else that needs updating?

So this is confusing, but actually makes sense. Bear with me:

- `npx . new` will run the `bin.js` file (which is tested by `bin.test.js`). 
- This file runs the CLI tool and assumes you want to clone the latest version. You can change that by passing the `-v` flag, for example `-v master`
- In this case, running `npx .` will run the `bin` file as per the current branch, but it will still try to clone the box as it was on the latest release. If you've changed anything in the box (ex. added a new dependency), it won't know, it will always clone the box as per the latest release.
- In this case in order to test, you want it to clone the box as per this branch. Once it's merged and gets into a new release, that will be the version the user will get when he runs `npx aztec-app`

So to answer your question, you need to pass this branch in order to get the box exactly as it will look like once merged. You can do that with the very same `-v` flag: `npx . new -v zpedro/boxes_readme`. You may need to adjust the `@aztec` dependencies to `latest`.

signorecello committed:
this idiot of yours had removed package.json from the wrong box...

signorecello commented:
Just pushed the change to make sure the `init` command now **won't** add useless `package.json`:

<img width="972" alt="Screenshot 2024-09-06 at 11 02 55" src="https://github.com/user-attachments/assets/4ae55d2d-18a1-4382-84c6-822faa60f644">
